rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -570,6 +560,24 @@ public String getParentShortHash() {
         return new byte[0];
     }
 
+    /**
+     * Compute the base hash for merged mining, taking into account whether the block is a umm block.
+     * This base hash is later modified to include the forkdetectiondata in its last 12 bytes
+     *
+     * @return The computed hash for merged mining
+     */
+    private byte[] getBaseHashForMergedMining() {
+        byte[] encodedBlock = getEncoded(false, false);
+        byte[] hashForMergedMining = HashUtil.keccak256(encodedBlock);
+
+        if (isUMMBlock()) {
+            byte[] leftHash = Arrays.copyOfRange(hashForMergedMining, 0, 20);
```

Comment:
Could we use UMM_LEAVES_LENGTH instead of magic number 20?

Reply:
Done